### PR TITLE
[codex] Add Groq thinking support

### PR DESCRIPTION
**Pull request by Taewan-P**
## Summary
- add a dedicated Groq API client and Groq-specific request/response DTOs instead of routing Groq through `OpenAIAPIImpl`
- route `ClientType.GROQ` through a new Groq repository flow that prefers structured reasoning fields and falls back to raw `<think>` parsing
- add parser and repository tests covering parsed reasoning, raw tag fallback, empty think blocks, and Groq request settings
- Resolves #261  

## Root Cause
Groq was sharing the generic OpenAI-compatible chat-completions path, which only read `delta.content`. That meant Groq-specific reasoning output was never mapped into `ApiState.Thinking`, so `<think>` content either stayed inline with the answer or produced blank spacing.

## Validation
- `GRADLE_USER_HOME=/tmp/gradle-home ./gradlew testDebugUnitTest`

## Summary by CodeRabbit

* **New Features**
  * Groq API support with streaming chat completions and configurable request parameters
  * Reasoning extraction: background "thinking" is parsed and emitted separately from visible answers

* **Improvements**
  * Better streaming/network error handling and timeout behavior for Groq requests

* **Tests**
  * New unit and integration tests covering reasoning parsing and Groq chat flows

### CODE REVIEW

File: app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/dto/groq/request/GroqChatCompletionRequest.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@Serializable
+data class GroqChatCompletionRequest(
+    @SerialName("model")
+    val model: String,
+
+    @SerialName("messages")
+    val messages: List<ChatMessage>,
+
+    @SerialName("stream")
+    val stream: Boolean = true,
```
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/dto/groq/request/GroqChatCompletionRequest.kt:20]: Added `@EncodeDefault(EncodeDefault.Mode.ALWAYS)` to `stream`. The shared serializer currently has `encodeDefaults = true`, but making the request model explicit avoids relying on the global setting for streaming behavior.

File: app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/network/GroqAPIImpl.kt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                    emit(NetworkClient.openAIJson.decodeFromString<GroqChatCompletionChunk>(response.body()))
+                    return@execute
+                }
+
+                val channel = response.bodyAsChannel()
+                while (!channel.isClosedForRead) {
+                    val line = channel.readLine() ?: break
+                    if (!line.startsWith("data: ")) continue
+
+                    val data = line.removePrefix("data: ").trim()
```
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/network/GroqAPIImpl.kt]: Adjusted SSE parsing to accept both `data: ...` and `data:...` forms by checking `startsWith("data:")` and trimming after `removePrefix`. That keeps the Groq stream parser tolerant of proxy or implementation differences.

File: app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/GroqReasoningParser.kt [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+                val safeLength = pendingContent.length - partialSuffixLength(
+                    pendingContent,
+                    GROQ_THINK_CLOSE
+                )
+                if (safeLength <= 0) break
+
+                pendingThought.append(pendingContent.substring(0, safeLength))
+                pendingContent.delete(0, safeLength)
+                break
```
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/GroqReasoningParser.kt:88]: Verified this against the current parser behavior and I’m not changing it. Raw `<think>` handling is only the fallback when Groq does not send structured reasoning; buffering until `</think>` closes is what lets us suppress whitespace-only thought blocks, trim the answer gap correctly, and avoid leaking malformed partial tags into the UI. Structured `delta.reasoning` and `message.reasoning` paths still stream incrementally.

File: app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImpl.kt [hunk 4]
```diff
@@ -801,6 +873,26 @@ class ChatRepositoryImpl @Inject constructor(
...
+
+    return GroqChatCompletionRequest(
+        model = platform.model,
+        messages = messages,
+        stream = platform.stream,
+        temperature = platform.temperature,
+        topP = platform.topP,
+        reasoningEffort = if (usesGptOssReasoning) "medium" else null,
+        reasoningFormat = if (platform.reasoning && !usesGptOssReasoning) "parsed" else null,
+        includeReasoning = if (usesGptOssReasoning) true else null
```
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImpl.kt]: Fixed in `createGroqChatCompletionRequest`: GPT-OSS requests now send `include_reasoning=false` when the toggle is off, and `ChatRepositoryImplTest` now asserts that explicit disable value.

File: app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImpl.kt [hunk 5]
```diff
@@ -801,6 +873,26 @@ class ChatRepositoryImpl @Inject constructor(
...
+    val usesGptOssReasoning = platform.reasoning && isGroqGptOssModel(platform.model)
+
+    return GroqChatCompletionRequest(
+        model = platform.model,
+        messages = messages,
+        stream = platform.stream,
+        temperature = platform.temperature,
+        topP = platform.topP,
+        reasoningEffort = if (usesGptOssReasoning) "medium" else null,
+        reasoningFormat = if (platform.reasoning && !usesGptOssReasoning) "parsed" else null,
```
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImpl.kt]: Fixed in `createGroqChatCompletionRequest`: non-GPT-OSS Groq requests now send `reasoning_format="hidden"` when reasoning is disabled, with repository coverage in `ChatRepositoryImplTest`.

File: app/src/test/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImplTest.kt [hunk 6]
```diff
@@ -29,21 +60,243 @@ class ChatRepositoryImplTest {
...
+                },
+                stream = {
+                    flowOf(ApiState.Success("done"))
+                }
+            ).first()
+        }
+        val elapsedMillis = System.currentTimeMillis() - startedAt
+
+        assertEquals(ApiState.Loading, firstState)
+        assertTrue(elapsedMillis < 150)
```
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/test/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImplTest.kt]: Removed the elapsed-time threshold. The test now only asserts that `Loading` is the first emission within the timeout window, which keeps the intent without the CI-flaky wall-clock check.

File: app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImpl.kt [hunk 7]
```diff
@@ -801,6 +873,26 @@ class ChatRepositoryImpl @Inject constructor(
...
+    platform: PlatformV2
+): GroqChatCompletionRequest {
+    val usesGptOssReasoning = platform.reasoning && isGroqGptOssModel(platform.model)
+
+    return GroqChatCompletionRequest(
+        model = platform.model,
+        messages = messages,
+        stream = platform.stream,
+        temperature = platform.temperature,
+        topP = platform.topP,
```
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImpl.kt:888]: Verified this one against the current Groq path and left it as-is. Groq accepts `temperature` and `top_p` alongside reasoning settings, and keeping the user-configured sampling values preserves current Groq behavior. I didn’t find a failing case or API constraint that would justify changing these fields in this PR.

File: app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImpl.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/main/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImpl.kt]: Fixed alongside the reasoning-toggle update: non-GPT-OSS requests now send `reasoning_format="hidden"`, GPT-OSS requests send `include_reasoning=false`, and the tests cover both cases.

File: app/src/main/kotlin/dev/chungjungsoo/gptmobile/di/NetworkModule.kt [hunk 8]
```diff
@@ -26,6 +28,10 @@ object NetworkModule {
     @Singleton
     fun provideOpenAIAPI(networkClient: NetworkClient): OpenAIAPI = OpenAIAPIImpl(networkClient)
 
+    @Provides
+    @Singleton
+    fun provideGroqAPI(networkClient: NetworkClient): GroqAPI = GroqAPIImpl(networkClient)
```
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/main/kotlin/dev/chungjungsoo/gptmobile/di/NetworkModule.kt:33]: Fixed by removing mutable request configuration from `GroqAPIImpl`. `ChatRepositoryImpl` now passes `token` and `apiUrl` into `streamChatCompletion(...)` per request, so the singleton no longer carries cross-request state.

File: app/src/test/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImplTest.kt [hunk 9]
```diff
@@ -1,12 +1,43 @@
 package dev.chungjungsoo.gptmobile.data.repository
 
+import android.content.ContextWrapper
+import dev.chungjungsoo.gptmobile.data.database.dao.ChatPlatformModelV2Dao
+import dev.chungjungsoo.gptmobile.data.database.dao.ChatRoomDao
+import dev.chungjungsoo.gptmobile.data.database.dao.ChatRoomV2Dao
+import dev.chungjungsoo.gptmobile.data.database.dao.MessageDao
+import dev.chungjungsoo.gptmobile.data.database.dao.MessageV2Dao
```
[the remark this answers is not in the document]
Comment (Taewan-P) [reply app/src/test/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImplTest.kt]: Removed the unused DAO imports from `ChatRepositoryImplTest` and verified the updated code with `./gradlew testDebugUnitTest`.

File: app/src/test/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImplTest.kt [hunk 10]
```diff
@@ -29,21 +54,255 @@ class ChatRepositoryImplTest {
...
+
+    @Test
+    fun `groq raw think fallback populates thinking state`() = runBlocking {
+        val groqAPI = FakeGroqAPI(
+            flowOf(
+                GroqChatCompletionChunk(
+                    choices = listOf(
+                        GroqChoice(
+                            index = 0,
+                            delta = GroqDelta(content = "<think>Secret</think>\nVisible")
```
Comment (rsb-23) [inline app/src/test/kotlin/dev/chungjungsoo/gptmobile/data/repository/ChatRepositoryImplTest.kt:121]: Maybe it fails for `Secret\n\nWord` within think tag.
